libs/community/langchain_community/retrievers/azure_ai_search.py

```diff
@@ -52,9 +54,14 @@ def validate_environment(cls, values: Dict) -> Dict:
         values["index_name"] = get_from_dict_or_env(
             values, "index_name", "AZURE_AI_SEARCH_INDEX_NAME"
         )
-        values["api_key"] = get_from_dict_or_env(
-            values, "api_key", "AZURE_AI_SEARCH_API_KEY"
-        )
+        try:
+            values["api_key"] = get_from_dict_or_env(
+                values, "api_key", "AZURE_AI_SEARCH_API_KEY"
+            )
+        except ValueError:
+            values["azure_ad_token_provider"] = get_from_dict_or_env(
```

Comment:
How did you decide the order of things to check? i.e. how did you decide to check api_key first versus azure_ad_token_provider?

Reply:
I didn't think too much about this decision. My assumption was that most people will use keys compared to MI. Wasn't sure as well how to handle the checking for the api_key env variable without try/except. Open to other suggestions!

Reply:
Let me actually change this into an if else statement and change the order to first check for the token and the else out to api key. This is consistent with what we're doing later, so makes sense to change the order.

Reply:
@pamelafox changed this code to use if/else and check to see if the token provider is given in the values dictionary first. I think this works because it's rare for people to set the token provider as an env variable.